viewer/src/datamodels/cad/CadModelUpdateHandler.ts

```diff
@@ -71,14 +73,43 @@ export class CadModelUpdateHandler {
         );
       }),
       // Load sectors from repository
-      publish((wantedSectors: Observable<WantedSector[]>) =>
-        wantedSectors.pipe(
-          switchAll(),
-          distinctUntilLevelOfDetailChanged(),
+      publish((wantedSectors: Observable<WantedSector[]>) => {
+        const modelSectorStates: { [blobUrl: string]: { [id: number]: LevelOfDetail } } = {};
+        const unloadedSectorsObservable = wantedSectors.pipe(
+          flatMap(wantedSectorArray => {
+            return from(wantedSectorArray).pipe(
+              filter(wantedSector => {
+                try {
+                  const sectorStates = modelSectorStates[wantedSector.blobUrl];
+                  const sectorState = sectorStates[wantedSector.metadata.id];
+                  return !(sectorState && sectorState === wantedSector.levelOfDetail);
+                } catch (error) {
+                  return true;
+                }
```

Comment:
Why do we need to hold state? And when will this fail (i.e. error is catched and true is returned)?

Reply:
true means that it will be requested. In this scenario if say the model doesn't exist and is undefined it will ask for it. Further down it does the creation of the "model" in the state if it is undefined.

Reply:
I think we should avoid catching error like this, and rather check for undefined and return true in that case.

Reply:
Sorted